src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -129,10 +154,29 @@ export const NewProposal = ({ navigation, data, query }) => {
         tagList: newProposalData?.tagList,
         termsOfService: hasAcceptedTermsOfService,
         videoUrl: newProposalData?.videoUrl
+
+        // TODO: image and document upload needed here? and if yes, how to do it?
+        //  documentsAttributes: [
+        // 	  {
+        // 		  title: 'sample.pdf',
+        // 		  cachedAttachment:
+        // 		  	'/Users/ardasenturk/Development/SVA/consul-bb/public/system/documents/cached_attachments/user/49/original/f2245afb48be5f906474ad929aacb7f91f408a23.pdf'
+        // 	  }
+        //  ],
       }
     };
 
-    if (!hasAcceptedTermsOfService) return showRegistrationFailAlert();
+    if (!hasAcceptedTermsOfService) return showPrivacyCheckRequireAlert();
+
+    if (newProposalData.image) {
+      const imageAttributes = await uploadImage(newProposalData.image);
+
+      if (!imageAttributes) {
+        return;
+      } else {
+        variables.attributes = { ...variables.attributes, imageAttributes };
```

Comment:
wouldn't this work in any case? why does this need to be an if-else with a return case?

Reply:
If the image upload fails, the proposal opens without the image. I had to check for this. If the image upload is unsuccessful, it should notify the user and then change this image.

Reply:
0bc389056155a1878daf189d621c7353bcc70657